File: pset1/mario.c
#include <cs50.h>
#include <stdio.h>

// Prototype "breadcrumbs"
void left_spaces(int n);
void blocks(int n);

int main(void)
{
    int n;
    do
    {
        n = get_int("Height: ");
    }
    // Checks user input is between 1 and 8
    while (n < 1 || n > 8);

    for (int i = 0; i < n; i++)
    {
        int j, k;

        // Variable to keep track of number of spaces
        j = (n - i) - 1;

        // Variable to keep track of number of blocks
        k = i + 1;

        left_spaces(j);
        blocks(k);
        printf("  ");
        blocks(k);
        printf("\n");
    }
}

// Will print left-aligned spaces
void left_spaces(int n)
{
    for (int i = 0; i < n; i++)
    {
        printf(" ");
    }
}

// Will print blocks as '#' character
void blocks(int n)
{
    for (int i = 0; i < n; i++)
    {
        printf("#");
    }
}

File: pset6/mario.py
# Python implementation of the mario-more problem set

import sys

while True:
    try:
        height = int(input("Height: "))
        if (height > 0) & (height < 9):
            for i in range(height):
                # Variable to keep track of number of spaces
                j = (height - i) - 1

                # Variable to keep track of number of blocks
                k = i + 1
                print(j*" "+k*"#"+"  "+k*"#")
            break
    except Exception as e:
        continue